This is my CSC596 proposal for the Fall of 2019. AH (#19)
compusciencing/compusciencing.github.io: _posts/2019-04-01-develop-implement-optimize-neural-nets.md

---
layout: post
title: "Design, Implementation, and Optimization of Neural Networks Proposal"
categories: ["Missouri State University", "proposal"]

> tags: ["CSC 596", "AI", "Neural Networks", "C++"]

Reviewer: They should all be lowercase, and the first one should be "csc596."

author: "Anthony Harris"
---

> Preface: This project will focus on the developement, implementation, and optimization of [Artificial Neural Networks](https://www.youtube.com/watch?v=aircAruvnKk&list=PLZHQObOWTQDNU6R1_67000Dx_ZCJB-3pi) in at least 2 sub projects of the [Omega Gaming Project](https://www.omega-gaming-project.org/). The basic structure of the artificial neural networks will be based on the principle ideas behind the AlphaNet design of AI to be implemented in the Omega Gaming Project. In short, the goal of the AlphaNet design is to minimize the amount of processing power required by a neural network while maximizing human-like information processing by the abstraction of data to a binary state where possible.

Reviewer: ```suggestion
This project will focus on the development, implementation, and optimization of [Artificial Neural Networks](https://www.youtube.com/watch?v=aircAruvnKk&list=PLZHQObOWTQDNU6R1_67000Dx_ZCJB-3pi) (ANNs) in at least 2 sub projects of the [Omega Gaming Project](https://www.omega-gaming-project.org/). I will base the ANNs on the principle ideas behind the AlphaNet design of AI, which will be implemented in the Omega Gaming Project. In short, the goal of the AlphaNet design is to minimize the amount of processing power required by a neural network while maximizing human-like information processing by the abstraction of data to a binary state where possible.
```

> Sub-Project 1: Initially, I will take an existing random word generator and determine whether or not a given word is legible using an artificial neural network. Then, I will move to optimize the neural net using different variants of the neural net and grading their fitness based on the "confidence" of the output divided by the time took to process. Given success, this also provides an opportunity to compare the results of individual variant optimization to the optimization methods found in the Neuro-Evolution of Augmenting Topologies, also known as N.E.A.T., method. Potential expansion of this sub-project might include the determination of legibility using different languages as a point of reference as opposed to just English.

Reviewer: ```suggestion
## Sub-Project 1

Initially, I will take an existing random word generator and determine whether a given word is *legible* using an ANN. Then, I will optimize different ANN variants and grade their fitness based on the “confidence” of the output and the time taken to process. Given success, this also provides an opportunity to compare the results of individual variant optimization to the optimization methods found in the Neuro-Evolution of Augmenting Topologies, also known as N.E.A.T., method. Potential expansion of this sub-project might include the determination of legibility using different languages as a point of reference as opposed to just English.
```

Add a link to the NEAT paper.

> Sub-Project 2: After demonstrating a sufficient level of accuracy in sub-project 1, I will move to the design, implementation, and optimization of a similarly structured neural network with the goal of applying continuous application in a simple tank game. Ideally, I'd expect to have the tank game fully functional by that point in time, but it is currently still in developement. In the event it is not finished yet, some time may be dedicated to the finalizing of the game, or the determination of a different sub-problem.

Reviewer: ```suggestion
## Sub-Project 2

After showing a sufficient level of accuracy in sub-project 1, I will move to the design, implementation, and optimization of a similarly structured neural network with the goal of applying continuous application in a simple tank game. Ideally, I’d expect to have the tank game fully functional by that point in time, but it is still in development. If it is not finished yet, I may dedicate time to the completing the game.
```

Describe the tank game.

Reviewer: Give the sub-projects more meaningingful names.

> Closing: The opportunity to have dedicated time to research these topics is invaluable to me as a computer science student. Not only does this give me the experience needed to be prepared for an AI-reliant workforce, but this also gives me experience with the abstraction of data into a form that is more universally applicable. I would go so far as to say that this research will be the foundation upon which the AlphaNet AI is built.

Reviewer: ```suggestion
## Closing

The opportunity to have dedicated time to research these topics is invaluable. Not only does this give me the experience needed to be prepared for an AI-reliant workforce, but this also gives me experience with the abstraction of data into a form that is more universally applicable. I would say this research will be the foundation upon which the AlphaNet AI is built.
```
